server.js

```diff
@@ -1,28 +1,108 @@
-import express from "express";
-import cors from "cors";
-
-// If you're using one of our datasets, uncomment the appropriate import below
-// to get started!
-// import avocadoSalesData from "./data/avocado-sales.json";
-// import booksData from "./data/books.json";
-// import goldenGlobesData from "./data/golden-globes.json";
-// import netflixData from "./data/netflix-titles.json";
-// import topMusicData from "./data/top-music.json";
-
-// Defines the port the app will run on. Defaults to 8080, but can be overridden
-// when starting the server. Example command to overwrite PORT env variable value:
-// PORT=9000 npm start
-const port = process.env.PORT || 8080;
-const app = express();
+import express from "express"
+import cors from "cors"
+import booksData from "./data/books.json"
+
+// This defines that default port for the app to run - but can be overridden. 
+const port = process.env.PORT || 8080
+const app = express()
 
 // Add middlewares to enable cors and json body parsing
-app.use(cors());
-app.use(express.json());
+app.use(cors())
+app.use(express.json())
 
-// Start defining your routes here
+// Defining of landingpage routes
 app.get("/", (req, res) => {
-  res.send("Hello Technigo!");
-});
+  const landingPage = {
+     Hello: 
+     "here is an api of random famous books",
+     Routes: [{
+       "/books": "Get whole array of books",
+       "/books/id/'id number'": "Get unique book by id",
+       "/books?id='id number'": "Get unique book by id",
+       "/books/author/'name of author": "Get list of books from particular author",
+       "/books?author='name of author": "Get list of books from particular author",
+       "/books/title/'title of book": "Get books by title",
+       "/books?title='title of book": "Get books by title",
+     }]
+   }
+  res.send(landingPage)
+})
+
+//Routes with query
+//query example in browser http://localhost:8080/books?title=galaxy&author=douglas adams&id=377
+app.get("/books", (req, res) => {
+  const { id, author, title } = req.query
+  
+  let allBooks = booksData
+  
+  if (id) {
+    allBooks = allBooks.filter((book) => book.bookID === +id)
```

Comment:
I think you can use the "find"-method here, since it is only one book that is coming back when we use their unique id 😊